src/core/Akka.Tests/IO/TcpIntegrationSpec.cs

```diff
@@ -211,20 +212,45 @@ public async Task BugFix_3021_Tcp_Should_not_drop_large_messages()
                 var actors = await x.EstablishNewClientConnectionAsync();
 
                 // create a large-ish byte string
-                var str = Enumerable.Repeat("f", 567).Join("");
-                var testData = ByteString.FromString(str);
+                const int length = 2064;
```

Comment:
1. Create 3 random messages that are all reasonably large
2. Transmit them sequentially
3. Validate that the received byte buffers all match the input messages, when laid out  sequentially in a new output buffer